artiq/dashboard/experiments.py

```diff
@@ -611,6 +616,30 @@ def resolve_expurl(self, expurl):
         else:
             raise ValueError("Malformed experiment URL")
 
+    def set_argument_value(self, expurl, arg_name, arg_value, **kwargs):
+        argument = self.submission_arguments.get(expurl, {}).get(arg_name)
+        if expurl in self.submission_arguments:
+            if argument is not None:
+                self.update_argument_state(argument, arg_value, **kwargs)
+                dock = self.open_experiments.get(expurl)
+                if dock is not None:
+                    dock.argeditor.update_argument_state(arg_name)
+            else:
+                logger.warning("No argument '%s' found", arg_name)
+        else:
+            logger.error("No experiment '%s' found", expurl)
+
+    def update_argument_state(self, argument, arg_value, **kwargs):
+        state = argument["state"]
+        if argument["desc"]["ty"] == "Scannable":
+            for key in kwargs:
+                 if key in state and key != "selected":
+                    state.update({"selected": arg_value, **kwargs})
```

Comment:
Alternatively just add the valid ones, since you iterate over `kwargs` anyways. This would also give you the option to make line 639 a warning instead of an error, since right now without the `raise` there would be a bug here. Note my usage of the word *option* here by the way.

Reply:
Updated the part as suggested.